src/Main/Main.jsx

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/jsx-key */
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable react/prop-types */
+
+
+import { Link } from 'react-router-dom'
+import { useEffect, useState } from 'react'
+import Home from '../assets/Home.svg'
+import PostWhatsOnYourMind from './Post'
+import Comment from './Comments'
+import ViewComment from './Viewcomments'
+
+function Main(props) {
+    const { dataGotten, setDataGotten  } = props
+    const [ contactData, setContactData ] = useState({})
+    const [initials, setInitials] = useState({})
+    const [ comments, setComments ] = useState([])
+
+
+
+    if(!dataGotten) {
+        return <p> loading....</p>
+    }
+
+
+    const getcontactName = (contactId) => {
+        fetch(`https://boolean-api-server.fly.dev/Usamaibrahim33/contact/${contactId}`)
+          .then((response) => response.json())
+          .then((data) => {
+       
+
+            // console.log(`${data.firstName[0]}${data.lastName[0]}`)
+            setContactData((prevData) => ({
+                ...prevData,
+                [contactId]: `${data.firstName} ${data.lastName}`
```

Comment:
I'd be careful to override the contact Id which is usually an integer with a string. Better store it in the state for the initials!

Reply:
@Usamaibrahim33 I meant that you create a key named "contactId" with the name in it which could confuse you, because the contact ID is usually a number in the API. maybe better name it "full name" or something.